### ISSUE TITLE
GetHistoricalStats (account or character) endpoints will not return Gambit/PvECompetitive stats for now

### ISSUE THREAD

**Issue by vthornheart-bng** [Bungie-net/api#651]
Just discovered a bug that was preventing this subset of stats from being returned - unfortunately, the bug is in an underlying system that will take longer to fix.  I don't have an ETA yet, but for now GetHistoricalStatsForAccount will still return only AllPvP and AllPvE, and if you request any Gambit or PvECompetitive modes for GetHistoricalStats it will throw an error.

I will keep you all posted here when I find out more about the fix and when it may be able to go live.

Tracked by TFS 707561

**Comment by vthornheart-bng** [comment]
Releasing fix for this one on the 12th!

**Comment by bladefist** [comment]
@vthornheart-bng will this be out today?

**Comment by vthornheart-bng** [comment]
Ah, my apologies - I should have updated this bug.  We found that this is interacting with the root cause of #664 as well.  My initial fix is going to be insufficient - it went out yesterday, but the actual issue won't be fixed without the game-side fix for #664 that will come on the 28th.  My apologies for not updating this bug earlier!

**Comment by bladefist** [comment]
@vthornheart-bng Understood thanks.  While I have you, will the PGCR for Gambit have data in `teams` so we can separate who is on which team?

**Comment by vthornheart-bng** [comment]
Ah, this is the first I've heard of it - indeed, this slipped under our radar, that's a separate issue that I'm going to have to investigate.  I'll make a separate bug for that.

**Comment by bladefist** [comment]
Thanks!

**Comment by vthornheart-bng** [comment]
Filed as #679

**Comment by t3rr11** [comment]
It's not only gambit and pve competitive that returns errors. When requesting AllTime[2] Leaderboards[101], Raids[4], Strikes[3], Nightfalls[16] and quite a few others. They all return 1618.

**Comment by vthornheart-bng** [comment]
Hmm, interesting - I could indeed see this same root cause potentially happening for other activity modes, I'll need to do some investigating to see if the fix on the 18th will affect those as well.

**Comment by Jakosaur** [comment]
Will GetHistoricalStatsForAccount start showing Gambit stats from today or another time? Thanks

**Comment by vthornheart-bng** [comment]
It should start showing both old and new stats once our deployment is complete, around 3pm!  The fix should work retroactively.

**Comment by Jakosaur** [comment]
GetHistoricalStatsForAccount still doesn't seem to be showing Gambit/PvECompetitive for me 🤔

**Comment by vthornheart-bng** [comment]
Ah, just noticed this comment - I'll give it a look.

**Comment by heymrcarter** [comment]
I'm still not seeing Gambit stats in the GetHistoricalStats response. 
<img width="1412" alt="screen shot 2018-11-01 at 12 35 18 pm" src="https://user-images.githubusercontent.com/1762068/47865455-a53e6300-ddd2-11e8-91b7-cecbf185fae8.png">

This was a request to `/Destiny2/1/Account/4611686018458918179/Character/0/Stats/`

**Comment by vthornheart-bng** [comment]
Ah, I see what's going on here - it looks like the default results, if you don't specifically ask for Gambit/AllPvECompetitive, is not returning those stats.

I'll look into why that is, but for the time being you can work around this by explicitly requesting the modes you want returned using the "modes" querystring parameter!

**Comment by vthornheart-bng** [comment]
I'm going to add AllPvE and AllPvECompetitive to the "default" results, to match with the fact that we pass AllPvP when you don't ask for specific modes - but if anyone wants additional filtering you'll have to use the modes querystring parameter to get the specific modes you want!

**Comment by heymrcarter** [comment]
Awesome! thanks for your help and quick reply @vthornheart-bng

**Comment by vthornheart-bng** [comment]
Fixed as of v2.3.3 deployment (12/11)!
